tasks/install_shell.sh

```diff
@@ -810,19 +840,29 @@ case $platform in
     download_url="${apt_source}/${filename}"
     ;;
   "mac_os_x")
-    info "Mac platform! Lets get you a DMG..."
-    filetype="dmg"
+    arch="x86_64"
+    if [[ $(uname -p) == "arm" ]]; then
+        arch="arm64"
+    fi
     if test "$version" = "latest"; then
       filename="puppet-agent-latest.dmg"
     else
       filename="puppet-agent-${version}-1.osx${platform_version}.dmg"
     fi
+    info "Mac platform! Lets get you a DMG...!!"
+    if [[ "$collection" =~ core ]]; then
+      if [ -z "$password" ]; then
+      echo "A password parameter is required to install"
+      exit 1
```

Comment:
Since we check for an empty password earlier, is the `if [ -z "$password ]` check needed?